### PR TITLE
Create the project based on kubebuilder

### PR DESCRIPTION
**Pull request by merlintang**
Use kubebuilder to build xgboost operator

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/xgboost-operator/15)

### CODE REVIEW

File: config/rbac/rbac_role.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
Comment (terrytangyuan) [inline config/rbac/rbac_role.yaml]: Where do you need ClusterRole?

File: cover.out [hunk 2]
```diff
@@ -0,0 +1 @@
+mode: set
```
Comment (terrytangyuan) [inline cover.out]: Delete unnecessary files

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,55 @@
+
+# Image URL to use all building/pushing image targets
+IMG ?= controller:latest
+
+all: test manager
+
+# Run tests
+test: generate fmt vet manifests
+	go test ./pkg/... ./cmd/... -coverprofile cover.out
```
Comment (terrytangyuan) [inline Makefile]: Can the auto-generated tests run successfully? Let’s include them in Travis

**Comment by merlintang** [comment]
we want to build this project based on kubebuilder 2.0.x, thus, close this and resubmit other pr based on kubebuilder 2.0.x

**Pull request closed without merging**